src/Swift.Bindings/src/TypeDatabase/TypeRecord.cs

```diff
@@ -5,6 +5,29 @@
 
 namespace BindingsGeneration;
 
+/// <summary>
+/// Represents type's flags.
+/// </summary>
+[Flags]
+public enum TypeRecordFlags
+{
+    None = 0,
+    Frozen = 1 << 0,
+    HeapAllocated = 1 << 1,
```

Comment:
What does it mean that a type is heap allocated? That it is allocated on heap in swift?

Reply:
HeapAllocated flag indicates the type is heap allocated (class) or contains a heap allocated property (struct with a ref property). (Added a comment)

Reply:
I see, thanks! Could we encode this somehow into the name of this flag? Like `requiresMemoryManagement` or something similar?